tests/migrations/test_autodetector.py

```diff
@@ -2454,3 +2454,28 @@ def test_mti_inheritance_model_removal(self):
         self.assertNumberMigrations(changes, 'app', 1)
         self.assertOperationTypes(changes, 'app', 0, ['DeleteModel'])
         self.assertOperationAttributes(changes, 'app', 0, 0, name='Dog')
+
+    def test_base_field_removal_added_into_sub_model(self):
+        """
+        # 31416 - base fields removals take place before newly added
+        inheriting models with removed base fields added.
+        """
+        readable = ModelState("app", "readable", [
+            ("id", models.AutoField(primary_key=True)),
+            ("title", models.CharField(max_length=200)),
+            ("name", models.CharField(max_length=200)),
```

Comment:
Is it necessary to test with two fields? Feels like a single one would be enough to reproduce?

Reply:
i just wanted to test if multiple fields would work as well. Yes, a single one would be enough to reproduce the same error indeed.